app/Http/Controllers/Settings/MultiFAController.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace App\Http\Controllers\Settings;
+
+use Google2FA;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Foundation\Auth\RedirectsUsers;
+use PragmaRX\Google2FALaravel\Support\Authenticator;
+
+class MultiFAController extends Controller
+{
+    use RedirectsUsers;
+
+    protected $redirectTo = '/settings/security';
+
+    /**
+     * Session var name to store secret code.
+     */
+    private const SESSION_TFA_SECRET = '2FA_secret';
+
+    /**
+     * Create a new authentication controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('web');
+    }
+
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function enableTwoFactor(Request $request)
+    {
+        //generate new secret
+        $secret = $this->generateSecret();
+
+        $user = $request->user();
+
+        //generate image for QR barcode
+        $imageDataUri = Google2FA::getQRCodeInline(
+            $request->getHttpHost(),
+            $user->email,
+            $secret,
+            200
+        );
+
+        $request->session()->put(self::SESSION_TFA_SECRET, $secret);
+
+        return view('settings.security.2fa-enable', ['image' => $imageDataUri, 'secret' => $secret]);
+    }
+
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function validateTwoFactor(Request $request)
+    {
+        $this->validate($request, [
+            'one_time_password' => 'required',
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "one_time_password" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)